src/qibolab/instruments/qm/controller.py

```diff
@@ -322,6 +297,15 @@ def register_pulses(self, configs: dict[str, Config], sequence: PulseSequence):
             acquisitions (dict): Map from measurement instructions to acquisition objects.
         """
         for channel_id, pulse in sequence:
+            if (
+                hasattr(pulse, "duration")
+                and isinstance(pulse.duration, float)
```

Comment:
This should be now enforced by the `Pulse` constructor (Pydantic will enforce the type)

Reply:
Even the condition above: all pulses have a duration:
https://github.com/qiboteam/qibolab/blob/e7a8d185ce57ea316b8bdba61fd913a7f6dc9d6b/src/qibolab/pulses/pulse.py#L28-L29

Reply:
Since it is enforced I removed the check in dbd35989b67325ed71831ceb373f47daeddc819c. I just noticed that `.is_integer()` is not available for `int` so I wanted to be sure.